tests/src/testelementsaslisteners.cpp

```diff
@@ -59,16 +59,15 @@ class DummySkill : public ResultTypeBase<ResultTypes::linearStress>
   // This returns a tuple functions to be registered
   template <typename MT, typename BC>
   void subscribeToImpl(BC& bc) {
+    using State = typename BC::State;
     if constexpr (std::same_as<MT, NonLinearSolverMessages>) {
-      using NLSState = typename BC::State;
-      underlying().subscribe(bc, [&](NonLinearSolverMessages message, const NLSState& state) {
+      underlying().subscribe(bc, [&](NonLinearSolverMessages message, const State& state) {
         this->updateState(message, state.domain, state.correction);
       });
     } else if constexpr (std::same_as<MT, UpdateMessages>) {
-      underlying().subscribe(bc, [&](UpdateMessages message, int val) { this->updateState(message, val); });
-      underlying().subscribe(bc, [&](UpdateMessages message) { this->updateState(message); });
+      underlying().subscribe(bc, [&](UpdateMessages message, State val) { this->updateState(message, val.domain); });
```

Comment:
```suggestion
      underlying().subscribe(bc, [&](UpdateMessages message, const State& val) { this->updateState(message, val.domain); });
```
Because `State` is `NonlinearSolverState`?